### PR TITLE
[Modules] Aligned the CMK implementation across current modules

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated AutomationAccount implementation
- Updaed Batch implementation
- Updated Registries implementation

| Pipeline |
| - |
| [Automation: AutomationAccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.automation.automationaccounts.yml) |
| [Batch: BatchAccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.batch.batchaccounts.yml) |
| [ContainerRegistry: Registries](https://github.com/Azure/ResourceModules/actions/workflows/ms.containerregistry.registries.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Batch/batchAccounts/.test/encr.parameters.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
```
Comment (eriqua) [inline modules/Microsoft.Batch/batchAccounts/.test/encr.parameters.json:2]: Is batchAccounts one of the modules requiring a separated test for cmk?

File: modules/Microsoft.Automation/automationAccounts/deploy.bicep [hunk 2]
```diff
@@ -169,17 +172,17 @@ resource automationAccount 'Microsoft.Automation/automationAccounts@2020-01-13-p
...
-    }
+    encryption: !empty(cMKKeyName) ? {
+      keySource: 'Microsoft.KeyVault'
+      identity: {
+        userAssignedIdentity: cMKUserAssignedIdentityResourceId
+      }
+      keyVaultProperties: {
+        keyName: cMKKeyName
+        keyVaultUri: cMKKeyVault.properties.vaultUri
+        keyVersion: !empty(cMKKeyVersion) ? cMKKeyVersion : last(split(cMKKeyVaultKey.properties.keyUriWithVersion, '/'))
```
Comment (eriqua) [inline modules/Microsoft.Automation/automationAccounts/deploy.bicep:183]: Properties look very similar to the DataFactory implementation. Could it be tested if `keyVersion: !empty(cMKKeyVersion) ? cMKKeyVersion : null` works in this case as well i.e. if also this resource automatically assumes latest if no version is provided?

File: modules/Microsoft.Automation/automationAccounts/deploy.bicep [hunk 3]
```diff
@@ -11,24 +11,17 @@ param location string = resourceGroup().location
...
-  'Microsoft.Automation'
-  'Microsoft.Keyvault'
-])
-param encryptionKeySource string = 'Microsoft.Automation'
+@description('Optional. The name of the customer managed key to use for encryption.')
+param cMKKeyName string = ''
 
-@description('Optional. The name of key used to encrypt data. This parameter is needed only if you enable Microsoft.Keyvault as encryptionKeySource.')
-param keyName string = ''
+@description('Optional. User assigned identity to use when fetching the customer managed key. If not provided, a system-assigned identity can be used - but must be given access to the referenced key vault first.')
```
Comment (eriqua) [inline modules/Microsoft.Automation/automationAccounts/deploy.bicep:20]: Double check if description should be updated here as well if no systemAssignedIdentity can actually be used

File: modules/Microsoft.Automation/automationAccounts/deploy.bicep (hunk 2 above)
Comment (MrMCake) [reply modules/Microsoft.Automation/automationAccounts/deploy.bicep:183]: I did, and no. It is very inconsistent how the Product-Groups implemented the CMK interface. Not providing a version was usually always the first thing I tried.

File: modules/Microsoft.Automation/automationAccounts/deploy.bicep (hunk 3 above)
Comment (MrMCake) [reply modules/Microsoft.Automation/automationAccounts/deploy.bicep:20]: Should work: https://docs.microsoft.com/en-us/azure/automation/automation-secure-asset-encryption

File: modules/Microsoft.ContainerRegistry/registries/deploy.bicep [hunk 4]
```diff
@@ -154,9 +154,12 @@ param cMKKeyVaultResourceId string = ''
 @description('Optional. The name of the customer managed key to use for encryption. Note, CMK requires the \'acrSku\' to be \'Premium\'.')
 param cMKKeyName string = ''
 
-@description('Conditional. User assigned identity to use when fetching the customer managed key. Note, CMK requires the \'acrSku\' to be \'Premium\'. Required if \'cMKeyName\' is not empty.')
+@description('Conditional. User assigned identity to use when fetching the customer managed key. Note, CMK requires the \'acrSku\' to be \'Premium\'. Required if \'cMKKeyName\' is not empty.')
 param cMKUserAssignedIdentityResourceId string = ''
 
+@description('Optional. The version of the customer managed key to reference for encryption. If not provided, the latest key version is used.')
```
[the remark this answers is not in the document]
Comment (MrMCake) [reply modules/Microsoft.ContainerRegistry/registries/deploy.bicep:157]: done

File: modules/Microsoft.Batch/batchAccounts/.test/encr.parameters.json (hunk 1 above)
Comment (MrMCake) [reply modules/Microsoft.Batch/batchAccounts/.test/encr.parameters.json:2]: Honestly, by now I can't tell you anymore :/

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**